addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js

```diff
@@ -279,8 +280,12 @@ export class AnalyticDistribution extends Component {
                     fields: analyticAccountFields,
                     activeFields: analyticAccountFields,
                 },
-                // company domain might be required here
-                domain: [["root_plan_id", "=", account.planId]],
+                domain: [
+                    ["root_plan_id", "=", account.planId],
+                    "|",
+                    ["company_id", "child_of", this.props.record.data[this.props.company_field][0]],
```

Comment:
In case the company is not set here, you'll get an `undefined`

Reply:
It doesn't work. When being on a branch, I don't have access to the analytic accounts from the parent company.

Reply:
Indeed this is not (and shouldn't be) a mandatory prop

Reply:
also keep in mind, someone might clear the company_id field

Reply:
['|', ('company_id', '=', False), ('company_id', 'parent_of', <company_id>)]

Reply:
after discussion: `['|', ('company_id', '=', False), ('company_id', '=', <company_id>)]`